framework/Optimizers/GradientDescent.py

```diff
@@ -0,0 +1,854 @@
+    self._functionalConstraintExplorationLimit = 500 # number of input-space explorations allowable for functional constraints
+    # __private
+    # additional methods
+    ## register adaptive sample identification criteria
+    self.registerIdentifier('purpose') # whether an opt, or which grad point
+
+  def handleInput(self, paramInput):
+    """
+      Read input specs
+      @ In, paramInput, InputData.ParameterInput, parameter specs interpreted
+      @ Out, None
+    """
+    RavenSampled.handleInput(self, paramInput)
+
+    # grad strategy
+    gradParentNode = paramInput.findFirst('gradient')
+    if len(gradParentNode.subparts) != 1:
+      self.raiseAnError('The <gradient> node requires exactly one gradient strategy! Choose from: ', gradKnownTypes())
+    gradNode = next(iter(gradParentNode.subparts))
+    gradType = gradNode.getName()
+    self._gradientInstance = gradReturnInstance(gradType, self)
+    self._gradientInstance.handleInput(gradNode)
+
+    # stepping strategy
+    stepNode = paramInput.findFirst('stepSize')
+    if len(stepNode.subparts) != 1:
+      self.raiseAnError('The <stepNode> node requires exactly one stepping strategy! Choose from: ', stepKnownTypes())
+    stepNode = next(iter(stepNode.subparts))
+    stepType = stepNode.getName()
+    self._stepInstance = stepReturnInstance(stepType, self)
+    self._stepInstance.handleInput(stepNode)
+
+    # acceptance strategy
+    acceptNode = paramInput.findFirst('acceptance')
+    if acceptNode:
+      if len(acceptNode.subparts) != 1:
+        self.raiseAnError('The <acceptance> node requires exactly one acceptance strategy! Choose from: ', acceptKnownTypes())
+      acceptNode = next(iter(acceptNode.subparts))
+      acceptType = acceptNode.getName()
+      self._acceptInstance = acceptReturnInstance(acceptType, self)
+      self._acceptInstance.handleInput(acceptNode)
+    else:
+      # default to strict mode acceptance
+      acceptNode = acceptReturnInstance('Strict', self)
+
+    # convergence options
+    convNode = paramInput.findFirst('convergence')
+    if convNode is not None:
+      for sub in convNode.subparts:
+        if sub.getName() == 'persistence':
+          self._requiredPersistence = sub.value
+        elif sub.getName() == 'terminateFollowers':
+          self._terminateFollowers = sub.value
+          self._followerProximity = sub.parameterValues.get('proximity', 1e-2)
+        elif sub.getName() == 'constraintExplorationLimit':
+          self._functionalConstraintExplorationLimit = sub.value
+        else:
+          self._convergenceCriteria[sub.name] = sub.value
+    if not self._convergenceCriteria:
+      self.raiseAWarning('No convergence criteria given; using defaults.')
+      self._convergenceCriteria['gradient'] = 1e-6
+    # same point is ALWAYS a criterion
+    self._convergenceCriteria['samePoint'] = 1e-16 # TODO user option?
```

Comment:
As the comment above it notes, this value is set without regard for the input currently but could be changed in the future to be a user option. Since we do not have a standard way to uniformly declare default tolerances across RAVEN, I recommend we move this suggestion to a new PR.

Reply:
As I suggested before, for this PR, It seems to me we should use the same value.

Reply:
This value is not related to obtaining a matching realization from a dataset, so there is no reason to force it to match values unless we have such a requirement throughout the code. This value behaves well in all the functional tests in which we use it across multiple machines.

I'm perfectly fine with a global tolerance value, but not adding it as part of this PR.